scoped-transitions.md

```diff
@@ -0,0 +1,110 @@
+    contain: layout;
+  }
+  .inner {
+    page-transition-tag: inner;
+    contain: layout;
+  }
+</style>
+<div class="outer">
+  <div class="inner"></div>
+</div>
+<script>
+  document.querySelector('.outer').createTransition(…);
+</script>
+```
+
+produces the following DOM tree under `outer`:
+
+```
+outer
+└─ ::page-transition
+   ├─ ::page-transition-container(outer)
+   │  └─ ::page-transition-image-wrapper(outer)
+   │     ├─ ::page-transition-outgoing-image(outer)
+   │     └─ ::page-transition-incoming-image(outer)
+   └─ ::page-transition-container(inner)
+      └─ ::page-transition-image-wrapper(inner)
+         ├─ ::page-transition-outgoing-image(inner)
+         └─ ::page-transition-incoming-image(inner)
+
+Note: We will probably need to change the naming, as 'page-transition' doesn't work.
+Note: We might want to add an ID for a transition to control which transition the `page-transition-tag` applies to.
+```
+
+The algorithm for executing the transition is as follows:
+
+1. At the next rendering opportunity after `createTransition`, the browser saves the painted output and geometry information for each tagged element under scoped-transition-root. The details are described in [7.3.4](https://drafts.csswg.org/css-shared-element-transitions-1/#perform-an-outgoing-capture-algorithm) except transforms are computed relative to the element. An async task is used to dispatch the `updateDOM` callback after this step.
+
+2. For every subsequent rendering opportunity, the browser renders the sub-tree under scoped-transition-root using the cached output captured in step 1. This is done by generating the pseudo-element sub-tree (originating from scoped-transition-root) as described in [7.9](https://drafts.csswg.org/css-shared-element-transitions-1/#create-transition-pseudo-elements-algorithm) except only outgoing-image pseudo-elements are generated. This allows the developer to update scoped-transition-root's subtree without presenting it to the user.
+
+   Note: Pseudo-elements at this stage don't need to be developer exposed so UAs can use an alternate approach for this step. Using pseudo-elements (and default browser CSS) here helps define a consistent behavior for snapshots are updated if the scoped-transition-root is resized.
+
+3. The captured output above is painted in a stacking context which is a descendant of scoped-transition-root's stacking context. The above example generates the following stacking context hierarchy.
+
+   ```
+   root
+   └─ outer
+      ├─ inner
+      └─ outer-page-transition (pseudo-tree underneath)
+   ```
+
+   If the scoped-transition-root is tagged, it's snapshot uses [content-capture](https://github.com/WICG/shared-element-transitions/blob/main/explainer.md#more-granular-style-capture) mode. This means that only the element's descendants (including both text and elements) are painted in the snapshot while box decorations (like border, box-shadow) continue to paint in the outer stacking context.
```

Comment:
I had the same thought, but worried it was weirdly different to what happens in root transitions. Also, are you proposing the full "granular style capture" model here? As in, what would happen if outer's background went from red to green. Would it animate (as in "granular style capture"), or switch (as in, we're just capturing the contents as a texture)?

Reply:
The idea comes from the comment you made. :) I thought a lot about it and realized it makes sense. If we try to capture the root snapshot (similar to root transitions) then it becomes difficult to describe the behaviour of clips + effects.

For example, scoped-transition-root has overflow:clip and a blur. The blur will draw outside the box because overflow:clip only affects the content (not self decorations). But the pseudo-element displaying scoped-transition-root's image will be its descendant. So it should clip the image and as a result will clip these box decorations.

> are you proposing the full "granular style capture" model here? As in, what would happen if outer's background went from red to green. Would it animate (as in "granular style capture"), or switch (as in, we're just capturing the contents as a texture)?

The latter, it'll be a direct flip. Its simpler because if the box resizes we can't avoid doing a relayout and repaint. And we'll have to teach the style/layout stack to use the old style values when painting these effects. If it's a direct flip, then style/layout/paint can proceed as usual. And way later we just skip drawing the tree under scoped-transition-root since the snapshot is drawing on top of it (same approach as today). Curious to hear @vmpstr's thoughts on this too.

Reply:
Added a section to the doc to capture this.